hypothesis/report
# Create a view with a similar report to Sparky

For:

 * https://github.com/hypothesis/report/issues/95

This is an attempt to replace Sparky with a job from Report.

If we can get this to work correctly, I think we can probably turn Sparky off, which might save us some money, and will definitely save us some headaches.

I think this is probably going to be much easier to adapt if people want changes (like showing various extra bits and bobs).

## Before merging

We need to work out if we can disable the 

## Testing notes

You can't really get a run of this working locally for a few reasons:

 * The local organization ids aren't in the Hubspot QA
 * Therefore we end up with no data
 * Even if you create data it won't line up so no rows are produced

To see the script running but not doing anything:

 * Run the create from scratch in H, LMS and Report
 * You should see a warning about no rows being inserted in Hubspot

You can patch together a better test by:

 * Copying the SQL from inside the view: https://github.com/hypothesis/report/blob/4ac220962053492a72d8aa6ef8215be5b23b5d91/report/sql_tasks/tasks/report/create_from_scratch/99_hubspot_write/01_company_property_update/01_create_view.sql#L5
 * Paste that into: https://report.hypothes.is/question/new
 * Download the result as a CSV file and put it in the same directory called `data.csv`
 * Comment out the return in the no rows check
 * Replace `csv_files=["report/sql_tasks/tasks/report/create_from_scratch/99_hubspot_write/01_company_property_update/data.csv"]`
 * Run: `tox -e dev --run-command "python bin/run_sql_task.py --task report/create_from_scratch/99_hubspot_write"`
 * Visit: https://app.hubspot.com/import/21081992
 * You should see a new entry
 * You should expect a large number of errors as the ids don't match between live and QA
 * You should see some successes though
 * You can go to: https://app.hubspot.com/contacts/21081992/company/<company_id> and see the same numbers as from the spreadsheet (if it was a non error row)
